### PR TITLE
Try adding a memory limit for the kotlin compiler

### PR DESCRIPTION
**Pull request by jmartinesp**
## Content

Tweak the heap sizes for both the Gradle Daemon and the Kotlin Compiler:

- The gradle daemon needs way more memory than the kotlin compiler.
- The kotlin compiler didn't seem to have a max value so it was probably taking more than it needed, causing resource starvation and either the gradle daemon to OOM or the CI runner to silently die.

## Motivation and context

I want to fix our CI failures caused by lack of RAM.

## Tests

I've merged some jobs that were split to avoid triggering this issue and run them > 5 times in a row. There didn't seem to be any memory issues so far, but they may still happen.

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request title will be used in the release note, it clearly define what will change for the user
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -39,20 +39,13 @@ jobs:
...
           ELEMENT_ANDROID_MAPTILER_DARK_MAP_ID: ${{ secrets.MAPTILER_DARK_MAP_ID }}
-        run: ./gradlew :app:assembleGplayDebug -PallWarningsAsErrors=true $CI_GRADLE_ARG_PROPERTIES
-      - name: Assemble debug Fdroid APK
-        if: ${{ matrix.variant == 'debug' }}
-        env:
-         ELEMENT_ANDROID_MAPTILER_API_KEY: ${{ secrets.MAPTILER_KEY }}
-         ELEMENT_ANDROID_MAPTILER_LIGHT_MAP_ID: ${{ secrets.MAPTILER_LIGHT_MAP_ID }}
-         ELEMENT_ANDROID_MAPTILER_DARK_MAP_ID: ${{ secrets.MAPTILER_DARK_MAP_ID }}
-        run: ./gradlew app:assembleFDroidDebug -PallWarningsAsErrors=true $CI_GRADLE_ARG_PROPERTIES
+        run: ./gradlew :app:assembleGplayDebug app:assembleFDroidDebug -PallWarningsAsErrors=true $CI_GRADLE_ARG_PROPERTIES
```
Comment (bmarty) [inline .github/workflows/build.yml:48]: We do not build F-Droid version anymore here?

File: .github/workflows/sonar.yml [hunk 2]
```diff
@@ -36,16 +36,8 @@ jobs:
...
-      - name: Build Fdroid Debug
-        run: ./gradlew :app:assembleFdroidDebug $CI_GRADLE_ARG_PROPERTIES
-      - name: Build Sample
-        run: ./gradlew :samples:minimal:assembleDebug $CI_GRADLE_ARG_PROPERTIES
-      - name: Build library fixtures
-        run: ./gradlew assembleDebug createFullJarDebugTestFixtures $CI_GRADLE_ARG_PROPERTIES
-      - name: Build app fixtures
-        run: ./gradlew :app:createFullJarGplayDebugTestFixtures $CI_GRADLE_ARG_PROPERTIES
+      - name: Build debug code and test fixtures
+        run: ./gradlew assembleDebug createFullJarDebugTestFixtures :app:createFullJarGplayDebugTestFixtures $CI_GRADLE_ARG_PROPERTIES
```
Comment (bmarty) [inline .github/workflows/sonar.yml:40]: I understand that minimal app is built with this?

**Review by bmarty — APPROVED** [review]
Just 2 questions, else SGTM, thanks!

File: .github/workflows/build.yml (hunk 1 above)
Comment (jmartinesp) [reply .github/workflows/build.yml:48]: We're building both ⬆️

File: .github/workflows/sonar.yml (hunk 2 above)
Comment (jmartinesp) [reply .github/workflows/sonar.yml:40]: It should, but I'd have to double check with a `--dry-run`.

Comment (jmartinesp) [reply .github/workflows/sonar.yml:40]: Confirmed, it does.

File: .github/workflows/build.yml (hunk 1 above)
Comment (bmarty) [reply .github/workflows/build.yml:48]: Ah yes, I read it too fast, sorry.

**Review by bmarty — APPROVED** [review]

**Pull request merged**